File: lesson_04/danielsonadjocy/app/src/main/python/prime.py
def is_prime(num):
    """
    Checks if num is a prime number.

    Args:
        num (int): The number to check.

    Returns:
        bool: True if num is prime, False otherwise.
    """
    if num <= 1:
        return False
    for val in range(2,int(num**(1/2))+1):
        if num % val == 0:
            return False
    return True

if __name__ == "__main__":
    print(4, is_prime(4))
    print(13, is_prime(13))
    # Checks each number between 2 and 100 to see if they're prime.
    for num in range(2, 101):
        print(str(num) +": "+ str(is_prime(num)))

File: lesson_04/danielsonadjocy/app/src/main/java/org/example/Prime.java
public class Prime {
  public static void main(String[] args) {
    System.out.println(4 + " " + isPrime(4));
    System.out.println(13 + " " + isPrime(13));

    // Checks each number between 2 and 100 to see if they're prime.
    for (int num = 2; num <= 100; num++) {
      System.out.println(num + ": " + isPrime(num));
    }
  }

  /**
   * Checks if num is a prime number.
   *
   * @param num the number to check
   * @return true if num is prime, false otherwise
   */
  public static boolean isPrime(int num) {
    if (num <= 1){
        return false;
    }
    for (int val = 2; val <= Math.sqrt(num); val++) {
      if (num % val == 0) {
        return false;
      }
    }
    return true;
  }
}
